### PR TITLE
Optimize object forces

### PR DESCRIPTION
**Pull request by D8H**

### CODE REVIEW

File: Core/GDCore/Extensions/Builtin/BaseObjectExtension.cpp [hunk 1]
```diff
@@ -617,6 +617,7 @@ void GD_CORE_API BuiltinExtensionsImplementer::ImplementsBaseObjectExtension(
           "number", ParameterOptions::MakeNewOptions())
       .MarkAsAdvanced();
 
+  // Deprecated
```
Comment (AlexandreSi) [inline Core/GDCore/Extensions/Builtin/BaseObjectExtension.cpp:620]: Should you hide it if it si deprecated?

**Review by AlexandreSi — APPROVED** [review]
LGTM

**Review by 4ian — APPROVED** [review]

File: GDJS/Runtime/runtimeobject.ts [hunk 2]
```diff
@@ -1344,7 +1346,20 @@ namespace gdjs {
...
-      this._forces.push(this._getRecycledForce(x, y, multiplier));
+      if (multiplier === 1) {
+        this._permanentForce.add(x, y);
+      } else if (
+        multiplier === 0 &&
+        this._instantForces.length > 0 &&
+        this._instantForces[0].getMultiplier() === 0
+      ) {
+        // Avoid to instantiate new a Force for each instance force.
+        this._instantForces[0].add(x, y);
```
Comment (4ian) [inline GDJS/Runtime/runtimeobject.ts:1357]: Nice. You could almost have had a `_permanentForce`, a `_instantForce` and a `_legacyForces`.

Comment (D8H) [reply GDJS/Runtime/runtimeobject.ts:1357]: Yes, it would be clearer, but it would take a bit more memory for objects with no force.

Comment (D8H) [reply GDJS/Runtime/runtimeobject.ts:1358]: We could maybe to something like a "ForceManager" with a lazy instantiation to reduce memory usage even more, but we can come back to this later.

Comment (4ian) [reply GDJS/Runtime/runtimeobject.ts:1358]: That's very true. Let's see later and it's true that avoiding more memory for each instance is a good thing to ensure.

**Pull request merged**